app/services/collators/gross_income_collator.rb

```diff
@@ -60,17 +62,46 @@ def monthly_student_loan
     def calculate_monthly_student_loan
       return 0.0 if categorised_income.key?(:student_loan)
 
-      if gross_income_summary.irregular_income_payments.exists?
+      if gross_income_summary.irregular_income_payments.student_loan.exists?
         total = 0
-        gross_income_summary.irregular_income_payments.each do |payment|
-          total += (payment.amount / 12)
+        gross_income_summary.irregular_income_payments.student_loan.each do |payment|
+          total += (payment.amount / irregular_payment_divisor(payment))
         end
         total
       else
         0.0
       end
     end
 
+    def monthly_unspecified_source_income
+      @monthly_unspecified_source_income ||= calculate_monthly_unspecified_source_income
+    end
+
+    def calculate_monthly_unspecified_source_income
+      return 0.0 if categorised_income.key?(:unspecified_source_income)
+
+      if gross_income_summary.irregular_income_payments.unspecified.exists?
+        total = 0
+        gross_income_summary.irregular_income_payments.unspecified.each do |payment|
+          total += (payment.amount / irregular_payment_divisor(payment))
+        end
+        total
+      else
+        0.0
+      end
+    end
+
+    def irregular_payment_divisor(irregular_income_payment)
+      case irregular_income_payment.frequency
+      when CFEConstants::ANNUAL_FREQUENCY
+        12
+      when CFEConstants::QUARTERLY_FREQUENCY
+        3
+      else
+        raise "Unprocessable irregular income payment frequency: #{irregular_income_payment.frequency}"
+      end
+    end
```

Comment:
If we validate the payment frequency at the model level and we specify valid values in the schema, do we need to worry about raising specific errors for invalid frequencies?

I assume our dear friend `SonarCloud` complains about case statements without a default, but maybe we could fetch values from a hash instead?

```ruby
IRREGULAR_PAYMENT_PERIODS = {
  CFEConstants::ANNUAL_FREQUENCY => 12,
  CFEConstants::QUARTERLY_FREQUENCY => 3,
}.freeze

def irregular_payment_divisor(payment)
  IRREGULAR_PAYMENT_PERIODS.fetch(payment)
end
```

I wonder whether it might be helpful to define a method on the irregular payment model, or even break the normalisation calculation into its own class?

Reply:
I've added a `#monthly_equivalent_amount` method to the IrregularIncomePayment class, on the grounds that "monthly equivalent" seems like a fairly commonly used term and it's such a simple bit of logic that it feels like a reasonably thing for the model to be able to know and communicate about itself.